Lmod-spider-no-hidden-cluster-modules.patch

```diff
@@ -1,17 +1,17 @@
 diff --git a/src/Spider.lua b/src/Spider.lua
-index 587ff3c1..84d08d95 100644
+index 095a8f44..ec89b33b 100644
 --- a/src/Spider.lua
 +++ b/src/Spider.lua
-@@ -971,8 +971,10 @@ function M._Level2(self, sn, entryA, possibleA)
-          for j = 1, #entryT.parentAA do
-             local parentA = entryT.parentAA[j]
-             for i = 1, #parentA do
--               b[#b+1] = parentA[i]
--               b[#b+1] = '  '
-+               if show_hidden or not parentA[i]:find("cluster/%.") then
-+                 b[#b+1] = parentA[i]
-+                 b[#b+1] = '  '
-+               end
+@@ -1282,8 +1282,10 @@ function M._Level2(self, sn, fullName, entryA, entryPA, possibleA, tailMsg)
```

Comment:
So, the patch does not use the extra arguments?

Reply:
No, it doesn't need them.

All we need in the bit we're patching is `parentA`.

We're basically just adding a guard that always hides hidden `cluster` modules.

Reply:
You can do this cleaner with hooks so you don't need to patch Lmod:

```lua

local function visible_hook(modT)
    -- modT is a table with: fullName, sn, fn and isVisible
    -- The latter is a boolean to determine if a module is visible or not

    if fullName:find("cluster/%.") then
        modT.isVisible = false
    end
end

```

Reply:
@wpoely86 Do you mind opening an issue on that?

I'll try and remember to include this in the next update...